packages/dd-trace/src/appsec/rule_manager.js

```diff
@@ -188,67 +113,31 @@ function concatArrays (files) {
   return [...files.values()].flat()
 }
 
-/*
-  ASM_DATA Merge strategy:
-  The merge should be based on the id and type. For any duplicate items, the longer expiration should be taken.
-  As a result, multiple Rule Data may use the same DATA_ID and DATA_TYPE. In this case, all values are considered part
-  of a set and are merged. For instance, a denylist customized by environment may use a global Rule Data for all
-  environments and a Rule Data per environment
-*/
-
-function mergeRulesData (files) {
-  const mergedRulesData = new Map()
-  for (const [, file] of files) {
-    for (const ruleData of file) {
-      const key = `${ruleData.id}+${ruleData.type}`
-      if (mergedRulesData.has(key)) {
-        const existingRulesData = mergedRulesData.get(key)
-        ruleData.data.reduce(rulesReducer, existingRulesData.data)
-      } else {
-        mergedRulesData.set(key, copyRulesData(ruleData))
+function extractErrors (obj) {
```

Comment:
worth a comment explaining what HOW it extract errors. I'm sure this is far from obvious for anyone that doesn't know exactly what the WAF output data looks like, even for me it's weird

Reply:
With the new implementation the logic is so much clear - it is not dealing with arbitrary objects anymore, but it looks for the relevant keys in the object to extract the errors. I think now there is no need to explain how it works. Let me know if you think so.